proxy/frontend/core/src/main/java/org/apache/shardingsphere/proxy/frontend/netty/CDCChannelInboundHandler.java

```diff
@@ -0,0 +1,148 @@
+        context.setStatus(CDCConnectionStatus.NOT_LOGGED_IN);
+        ctx.channel().attr(CONNECTION_CONTEXT_KEY).setIfAbsent(context);
+        Builder builder = CDCResponse.newBuilder();
+        builder.setServerGreetingResult(ServerGreetingResult.newBuilder().setServerVersion(ShardingSphereVersion.VERSION).setMaxProtocolVersion("1").setMinProtocolVersion("1").build());
+        ctx.writeAndFlush(builder.build());
+    }
+    
+    @Override
+    public void channelRead(final ChannelHandlerContext ctx, final Object msg) throws Exception {
+        CDCConnectionContext connectionContext = ctx.channel().attr(CONNECTION_CONTEXT_KEY).get();
+        CDCConnectionStatus status = connectionContext.getStatus();
+        CDCRequest request = (CDCRequest) msg;
+        if (CDCConnectionStatus.NOT_LOGGED_IN == status) {
+            processLogin(ctx, request);
+            return;
+        }
+        switch (request.getRequestCase()) {
+            case CREATE_SUBSCRIPTION:
+                processCreateSubscription(ctx, request);
+                break;
+            case START_SUBSCRIPTION:
+                processStartSubscription(ctx, request, connectionContext);
+                break;
+            case STOP_SUBSCRIPTION:
+                stopStartSubscription(ctx, request, connectionContext);
+                break;
+            case DROP_SUBSCRIPTION:
+                dropStartSubscription(ctx, request);
+                break;
+            case FETCH_RECORD_ACK:
+                break;
+            default:
+                log.warn("Cannot handle this type of request {}", request);
+        }
+    }
+    
+    private void processLogin(final ChannelHandlerContext ctx, final CDCRequest request) {
+        if (!request.hasLogin() || !request.getLogin().hasBasicBody()) {
+            ctx.writeAndFlush(CDCResponseGenerator.failed(request.getRequestId(), CDCResponseErrorCode.ILLEGAL_REQUEST_ERROR, "Miss login request body")).addListener(ChannelFutureListener.CLOSE);
+            return;
+        }
+        BasicBody body = request.getLogin().getBasicBody();
+        Collection<ShardingSphereRule> globalRules = ProxyContext.getInstance().getContextManager().getMetaDataContexts().getMetaData().getGlobalRuleMetaData().getRules();
+        Optional<AuthorityRule> authorityRule = globalRules.stream().filter(rule -> rule instanceof AuthorityRule).map(rule -> (AuthorityRule) rule).findFirst();
+        if (!authorityRule.isPresent()) {
+            ctx.writeAndFlush(CDCResponseGenerator.failed(request.getRequestId(), CDCResponseErrorCode.SERVER_ERROR, "Not find authority rule")).addListener(ChannelFutureListener.CLOSE);
+            return;
+        }
+        Optional<ShardingSphereUser> user = authorityRule.get().findUser(new Grantee(body.getUsername(), getHostAddress(ctx)));
+        if (user.isPresent() && Objects.equals(Hashing.sha256().hashBytes(user.get().getPassword().getBytes()).toString(), body.getPassword())) {
```

Comment:
Is sha256 result case-sensitive? It's better to uppercase or lowercase it, and also in CDC client

Reply:
Case-insensitive might be better, toLowerCase() or toUpperCase()